contracts/SimpleYieldEulerLever.sol

```diff
@@ -0,0 +1,395 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.14;
+
+import "@yield-protocol/utils-v2/contracts/cast/CastU128I128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastI128U128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I256.sol";
+import "@yield-protocol/utils-v2/contracts/token/TransferHelper.sol";
+import "./interfaces/IEulerMarkets.sol";
+import "./interfaces/IEulerEToken.sol";
+import "./YieldLeverBase.sol";
+
+/// @title A simple euler lever designed to work for one euler token & its underlying at a time
+/// @author iamsahu
+/// @notice Working:
+///         - Get flash loan of USDC/DAI/WETH
+///         - Deposit to get eulerToken
+///         - Deposit & borrow against it
+///         - Sell the fyToken to get USDC/DAI/WETH
+///         - Close the flash loan
+contract SimpleYieldEulerLever is YieldLeverBase {
+    using TransferHelper for IERC20;
+    using CastU128I128 for uint128;
+    using CastI128U128 for int128;
+    using CastU256U128 for uint256;
+    using CastU256I256 for uint256;
+
+    /// @notice euler market
+    IEulerMarkets public constant markets =
+        IEulerMarkets(0x3520d5a913427E6F0D6A83E07ccD4A4da316e4d3);
+    /// @notice Euler protocol address
+    address constant EULER = 0x27182842E098f60e3D576794A5bFFb0777E025d3;
+    /// @notice Asset underlying the Euler Token
+    address immutable asset;
```

Comment:
Would this be the asset that the user borrows from Yield? In that case we should call it `baseToken` to follow convention.